lib/zzhc/zzhc_port.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(zzhc_port, CONFIG_ZZHC_LOG_LEVEL);
+
+#include <zephyr.h>
+#include <base64.h>
+#include <secure_services.h>
+#include <json.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include "zzhc_internal.h"
+
+#define AT_PARAMS_MAX     11        /** Max. # of AT-params to parse */
+#define FICR_REV_OFFSET   0x134
+
+/** Cached storage of ICCID. */
+static char ueiccid[ICCID_LEN];
+
+/** Pointer to settings_handler. */
+static struct settings_handler settings_fn = {0};
+
+/** Name of settings */
+static char const settings_name[] = "zzhc";
+
+/**@brief Structure to store decoded json object. */
+struct rsp_obj {
+	const char *resultCode;
```

Comment:
Why camel case all of sudden, can we stick to the snake case with variables/struct members?

Reply:
In the adoption to Zephyr's json library, I have used `JSON_OBJ_DESCR_PRIM()`, which requires struct member names to be identical to the json name fields. To use snake case, `JSON_OBJ_DESCR_PRIM_NAMED()` has to be used.

In the next push, the struct member names will become snake case, and `JSON_OBJ_DESCR_PRIM_NAMED()` will be used instead.

Reply:
Done in the last push on 7th Nov, 2019 (ref: [here](https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1340#event-2777928390)).